ledger/src/blockstore_meta.rs

```diff
@@ -365,6 +412,51 @@ pub struct DuplicateSlotProof {
     pub shred2: shred::Payload,
 }
 
+#[derive(Deserialize, Serialize)]
+pub enum BlockStatus {
+    /// The block is being ingested, of the shreds received there are no conflicts
+    Incomplete,
+
+    /// The block `block_id` has been fully ingested, has consistent shreds and is ready for replay
+    Complete { block_id: Hash },
```

Comment:
Do we need an in between status for incomplete but in conflict, or is that automatically dumped?

Reply:
the blockstore should not care about this at some point consensus decides, at this point we could call a `cleanup()` function on blockstore to delete any other blocks in that slot